### ISSUE TITLE
IDISP003 should not warn here

### ISSUE THREAD

**Issue by JohanLarsson** [DotNetAnalyzers/IDisposableAnalyzers#175]
```cs
        private void UpdateRepositories()
        {
            foreach (var repository in this.allRepositories)
            {
                repository.Dispose();
            }

            this.allRepositories.Clear();
            if (this.gitDirectory is string path &&
                Directory.Exists(path))
            {
                foreach (var directory in Directory.EnumerateDirectories(path))
                {
                    if (Repository.TryCreate(directory, out var repository))
                    {
                        this.allRepositories.Add(repository);
                    }
                    else
                    {
                        foreach (var nested in Directory.EnumerateDirectories(directory))
                        {
                            if (Repository.TryCreate(nested, out repository))
                            {
                                this.allRepositories.Add(repository);
                            }
                        }
                    }
                }
            }
        }
```
